agent-shell.el

```diff
@@ -155,6 +217,14 @@ See `acp-make-initialize-request' for details."
   :type 'boolean
   :group 'agent-shell)
 
+(defcustom agent-shell-show-permission-diff-button t
```

Comment:
I'm not super sure what to call this just yet. How about something a little more generic like `agent-shell-offer-diffing`.